code/modules/shuttle/docking.dm

```diff
@@ -115,10 +115,8 @@
 		var/area/old_area = oldT.loc
 		var/move_mode = old_area.beforeShuttleMove(shuttle_areas) //areas
 
-		var/list/old_contents = oldT.contents
-		for(var/k in 1 to old_contents.len)
+		for(var/atom/movable/moving_atom as anything in oldT.contents)
```

Comment:
In reality the fix here is to not sleep during preflight check at all, but we do to try and reduce lag spikes. it just means there's the unfortunate low chance that some atoms will be missed

Reply:
budget constraints perhaps